pkg/loop/config_test.go

```diff
@@ -106,12 +137,66 @@ func TestEnvConfig_parse(t *testing.T) {
 					if config.TracingTLSCertPath != tc.expectedTracingTLSCertPath {
 						t.Errorf("Expected tracingTLSCertPath %s, got %s", tc.expectedTracingTLSCertPath, config.TracingTLSCertPath)
 					}
+					if config.TelemetryEnabled != tc.expectedTelemetryEnabled {
+						t.Errorf("Expected telemetryEnabled %v, got %v", tc.expectedTelemetryEnabled, config.TelemetryEnabled)
+					}
+					if config.TelemetryEndpoint != tc.expectedTelemetryEndpoint {
+						t.Errorf("Expected telemetryEndpoint %s, got %s", tc.expectedTelemetryEndpoint, config.TelemetryEndpoint)
+					}
+					if config.TelemetryInsecureConnection != tc.expectedTelemetryInsecureConn {
+						t.Errorf("Expected telemetryInsecureConn %v, got %v", tc.expectedTelemetryInsecureConn, config.TelemetryInsecureConnection)
+					}
+					if config.TelemetryCACertFile != tc.expectedTelemetryCACertFile {
+						t.Errorf("Expected telemetryCACertFile %s, got %s", tc.expectedTelemetryCACertFile, config.TelemetryCACertFile)
+					}
+					if !equalOtelAttributes(config.TelemetryAttributes, tc.expectedTelemetryAttributes) {
+						t.Errorf("Expected telemetryAttributes %v, got %v", tc.expectedTelemetryAttributes, config.TelemetryAttributes)
+					}
+					if config.TelemetryTraceSampleRatio != tc.expectedTelemetryTraceSampleRatio {
+						t.Errorf("Expected telemetryTraceSampleRatio %f, got %f", tc.expectedTelemetryTraceSampleRatio, config.TelemetryTraceSampleRatio)
+					}
+					if !equalStringMaps(config.TelemetryAuthHeaders, tc.expectedTelemetryAuthHeaders) {
+						t.Errorf("Expected telemetryAuthHeaders %v, got %v", tc.expectedTelemetryAuthHeaders, config.TelemetryAuthHeaders)
+					}
+					if config.TelemetryAuthPubKeyHex != tc.expectedTelemetryAuthPubKeyHex {
+						t.Errorf("Expected telemetryAuthPubKeyHex %s, got %s", tc.expectedTelemetryAuthPubKeyHex, config.TelemetryAuthPubKeyHex)
+					}
+					if config.TelemetryEmitterBatchProcessor != tc.expectedTelemetryEmitterBatchProcessor {
+						t.Errorf("Expected telemetryEmitterBatchProcessor %v, got %v", tc.expectedTelemetryEmitterBatchProcessor, config.TelemetryEmitterBatchProcessor)
+					}
+					if config.TelemetryEmitterExportTimeout != tc.expectedTelemetryEmitterExportTimeout {
+						t.Errorf("Expected telemetryEmitterExportTimeout %v, got %v", tc.expectedTelemetryEmitterExportTimeout, config.TelemetryEmitterExportTimeout)
+					}
 				}
 			}
 		})
 	}
 }
 
+func equalOtelAttributes(a, b OtelAttributes) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for k, v := range a {
+		if b[k] != v {
+			return false
+		}
+	}
+	return true
+}
+
+func equalStringMaps(a, b map[string]string) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	for k, v := range a {
+		if b[k] != v {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
Does testify have helpers for comparing map entries?

Reply:
Why not use `reflect.DeepEqual()`?

Reply:
Addressed https://github.com/smartcontractkit/chainlink-common/pull/967/commits/22280e07c68a8770b8e10ef4f0969553d63fd862